### ISSUE TITLE
No Ports Recognized with IDE 2.0.3 ARM64 build (Apple M1) on Mac Ventura

### ISSUE THREAD

**Issue by jeffmikels** [arduino/arduino-ide#1796]
### Describe the problem

Using the 2.0.3 build for native Apple Silicon works just fine on Monterey, but on Ventura no ports are recognized.

The workaround is to use 2.0.2 under Rosetta. I have not tested the 2.0.3 Intel build, so I don't know if the problem arose with 2.0.3 in general or with the Apple Silicon build specifically.

### To reproduce

- Install 2.0.3 ARM64 build on Mac Ventura
- Run it
- Plug in an Arduino Board
- No ports recognized in the IDE
- even though `ls /dev/cu.*` will correctly show the usb serial device

### Expected behavior

I would expect the IDE to discover the board or at least recognize the existence of the serial port.

### Arduino IDE version

2.0.3 ARM64

### Operating system

macOS

### Operating system version

Ventura (13.1)

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [X] My report contains all necessary details

**Comment by per1234** [comment]
Hi @jeffmikels. Thanks for your report. I am also using the Apple Silicon build of Arduino IDE 2.0.3 with macOS Ventura and the ports of my boards are shown in Arduino IDE, so the problem is not universal.

I think the best way to proceed with the investigation is to take a look at the logs generated by Arduino IDE to see whether there is a clue there. You can access the logs by running Arduino IDE from the command line.

Please do this:

1. Open your **Applications** folder.
1. Open **Utilities**.
1. Double-click on "**Terminal**".
1. Type the following command in the **Terminal** window:
   ```text
   "/Applications/Arduino IDE.app/Contents/MacOS/Arduino IDE"
   ```
1. Press **<kbd>Enter</kbd>**
1. Wait for Arduino IDE to finish starting.
1. Switch back to the **Terminal** window.
1. Press **<kbd>⌘</kbd>+<kbd>A</kbd>**.
   This will select all the text in the **Terminal** window.
1. Press **<kbd>⌘</kbd>+<kbd>C</kbd>**.
   This will [copy](https://lifehacker.com/how-to-copy-cut-and-paste-for-beginners-5801525) the selected text to the clipboard.
1. Click on the "**Leave a comment**" field here on this GitHub issue.
1. Press **<kbd>⌘</kbd>+<kbd>V</kbd>**.
   This will [paste](https://lifehacker.com/how-to-copy-cut-and-paste-for-beginners-5801525) the output into the code block.
1. Click the <kbd>**Comment**</kbd> button to make the reply.

---

Please let me know if you have any questions or problems while following those instructions.

**Comment by SauceCrusty** [comment]
I am also having this issue with my Arduino R3 Clone. It works fin in Windows 11 but on MAC Ventura on my M1 it doesn't show any ports. However it does show a serial port in System Report on the MAC when its plugged in.

**Comment by nickdesaulniers** [comment]
@jeffmikels @SauceCrusty try installing rosetta? See also #1818.

**Comment by PaulStoffregen** [comment]
FWIW, can not reproduce this problem with IDE 2.0.2 with AVR platform 1.8.3 (same as reported in this issue) with MacOS Ventura running on M1 Ultra, uploading to a genuine Arduino Leonardo board.

![Screenshot](https://user-images.githubusercontent.com/965463/212536329-abb20f63-5395-4b80-be31-06e8605e06b3.png)

**Comment by PaulStoffregen** [comment]
Same test works fine with IDE 2.0.3 (x86 version running with Rosetta)

![Screenshot](https://user-images.githubusercontent.com/965463/212536635-1466fd48-f229-4a59-9fd5-fe74bf0cc88c.png)

**Comment by PaulStoffregen** [comment]
And for completeness, also works fine with IDE 2.0.3 M1 version, this time with AVR platform 1.8.6 installed.

![Screenshot](https://user-images.githubusercontent.com/965463/212536819-375157dd-26da-479d-8d0f-806308fb4c49.png)

**Comment by SauceCrusty** [comment]
Not sure why or how but when I tried again I was able to see my ports and upload my sketch.

**Comment by per1234** [comment]
Closing due to lack of response to my question from 2 weeks ago, and because @SauceCrusty is no longer able to reproduce the fault.

@jeffmikels if you will provide the requested information by commenting here, I'll be happy to re-open this issue.

**Comment by jeffmikels** [comment]
I don't have access to a Ventura machine right now. 😞

**Comment by per1234** [comment]
OK, well if you are able to reproduce the issue again, please do comment here. I'll still be monitoring the thread.

The hypothesis offered at https://github.com/arduino/arduino-ide/issues/1796#issuecomment-1383038469 is that the [**serial-discovery**](https://github.com/arduino/serial-discovery) used by Arduino IDE to get a list of the serial ports available on your computer failed to start due to the **Rosetta 2** software not being installed on your Apple Silicon-based computer. **serial-discovery** is built for x86 hosts, which is not a problem when **Rosetta 2** is installed, but a no go without Rosetta. In this case, we would expect to see a "`bad CPU type in executable`" error in the logs you see when starting Arduino IDE from the command line.

**Comment by jhormigo** [comment]
I have a Ventura and exactly the same issue. Can't see the ports. I paste below what @per1234 requested:

`Last login: Sun Apr  9 20:33:04 on ttys006
(base) cy@mba13m2 ~ % cd ~/Applications/ArduinoCreateAgent
cd: no such file or directory: /Users/cy/Applications/ArduinoCreateAgent
(base) cy@mba13m2 ~ % cd ~/Applications/ArduinoCreateAgen
Session Contents Restored on 9 Apr 2023 at 21:31

Last login: Sun Apr  9 21:31:09 on console
(base) cy@mba13m2 ~ % ls /dev/cu.*
/dev/cu.Bluetooth-Incoming-Port
(base) cy@mba13m2 ~ % "/Applications/Arduino IDE.app/Contents/MacOS/Arduino IDE"
Arduino IDE 2.0.4
Starting backend process. PID: 2266
Using browser-only version of superagent in non-browser environment
Configuration directory URI: 'file:///Users/cy/.theia'
Configuring to accept webviews on '^.+\.webview\..+$' hostname.
2023-04-09T19:49:18.896Z root INFO Backend Object.initialize: 1.9 ms [Finished 0.355 s after backend start]
2023-04-09T19:49:18.896Z root INFO Backend Object.configure: 1.5 ms [Finished 0.355 s after backend start]
2023-04-09T19:49:18.896Z root INFO Backend MessagingContribution.onStart: 0.7 ms [Finished 0.355 s after backend start]
2023-04-09T19:49:18.897Z root INFO Backend OSBackendApplicationContribution.configure: 0.2 ms [Finished 0.371 s after backend start]
2023-04-09T19:49:18.897Z root INFO Backend DefaultWorkspaceServer.onStart: 0.1 ms [Finished 0.371 s after backend start]
2023-04-09T19:49:18.897Z root INFO Backend PluginLocalizationBackendContribution.configure: 0.1 ms [Finished 0.371 s after backend start]
2023-04-09T19:49:18.898Z root INFO Backend PluginLocalizationBackendContribution.initialize: 16.0 ms [Finished 0.371 s after backend start]
2023-04-09T19:49:18.898Z root INFO Backend TaskBackendApplicationContribution.onStart: 0.1 ms [Finished 0.371 s after backend start]
2023-04-09T19:49:18.898Z root INFO Backend ElectronTokenBackendContribution.configure: 0.5 ms [Finished 0.372 s after backend start]
2023-04-09T19:49:18.898Z config INFO >>> Initializing CLI configuration...
2023-04-09T19:49:18.898Z root INFO Backend PluginDeployerContribution.initialize: 0.2 ms [Finished 0.372 s after backend start]
2023-04-09T19:49:18.898Z root INFO Backend ConfigServiceImpl.onStart: 0.4 ms [Finished 0.372 s after backend start]
2023-04-09T19:49:18.899Z config INFO Loading CLI configuration from /Users/cy/.arduinoIDE/arduino-cli.yaml...
2023-04-09T19:49:18.899Z root INFO Backend WebviewBackendSecurityWarnings.initialize: 0.2 ms [Finished 0.373 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend ArduinoDaemonImpl.onStart: 0.5 ms [Finished 0.373 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend NodeFileUploadService.configure: 0.8 ms [Finished 0.373 s after backend start]
2023-04-09T19:49:18.899Z daemon INFO Starting daemon from /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/arduino-ide-extension/build/arduino-cli...
2023-04-09T19:49:18.899Z discovery-log INFO start
2023-04-09T19:49:18.899Z discovery-log INFO start new deferred
2023-04-09T19:49:18.899Z root INFO Backend HostedPluginLocalizationService.initialize: 0.2 ms [Finished 0.374 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend BoardDiscovery.onStart: 0.5 ms [Finished 0.374 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend FileDownloadEndpoint.configure: 0.4 ms [Finished 0.374 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend AuthenticationServiceImpl.onStart: 0.1 ms [Finished 0.375 s after backend start]
2023-04-09T19:49:18.899Z root INFO Backend PluginApiContribution.configure: 0.1 ms [Finished 0.375 s after backend start]
2023-04-09T19:49:18.900Z root INFO Backend HostedPluginReader.configure: 0.2 ms [Finished 0.375 s after backend start]
2023-04-09T19:49:18.900Z root INFO Backend HostedPluginReader.configure: 0.1 ms [Finished 0.375 s after backend start]
2023-04-09T19:49:18.900Z root INFO Backend PlotterBackendContribution.configure: 0.2 ms [Finished 0.375 s after backend start]
2023-04-09T19:49:18.901Z root INFO Theia app listening on http://localhost:53577.
2023-04-09T19:49:18.901Z root INFO Finished starting backend application: 1.6 ms [Finished 0.376 s after backend start]
2023-04-09T19:49:18.903Z root WARN The local plugin referenced by local-dir:/Users/cy/.arduinoIDE/plugins does not exist.
2023-04-09T19:49:18.903Z root WARN The local plugin referenced by local-dir:/Users/cy/.arduinoIDE/extensions does not exist.
2023-04-09T19:49:18.903Z root WARN The local plugin referenced by local-dir:/Users/cy/.arduinoProIDE/plugins does not exist.
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "cortex-debug" to a VS Code extension "cortex-debug@1.5.1" with engines:
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "vscode-arduino-tools" to a VS Code extension "vscode-arduino-tools@0.0.2-beta.7" with engines:
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "vscode-builtin-cpp" to a VS Code extension "cpp@1.52.1" with engines:
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "vscode-builtin-json" to a VS Code extension "json@1.46.1" with engines:
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "vscode-builtin-json-language-features" to a VS Code extension "json-language-features@1.46.1" with engines:
2023-04-09T19:49:18.920Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.920Z root INFO Resolved "vscode-language-pack-bg" to a VS Code extension "vscode-language-pack-bg@1.48.3" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-cs" to a VS Code extension "vscode-language-pack-cs@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-de" to a VS Code extension "vscode-language-pack-de@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-es" to a VS Code extension "vscode-language-pack-es@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-fr" to a VS Code extension "vscode-language-pack-fr@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-hu" to a VS Code extension "vscode-language-pack-hu@1.48.3" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-it" to a VS Code extension "vscode-language-pack-it@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-ja" to a VS Code extension "vscode-language-pack-ja@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-ko" to a VS Code extension "vscode-language-pack-ko@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-nl" to a VS Code extension "vscode-language-pack-nl@1.48.3" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-pl" to a VS Code extension "vscode-language-pack-pl@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-pt-BR" to a VS Code extension "vscode-language-pack-pt-BR@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-ru" to a VS Code extension "vscode-language-pack-ru@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-tr" to a VS Code extension "vscode-language-pack-tr@1.53.2" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-uk" to a VS Code extension "vscode-language-pack-uk@1.48.3" with engines:
2023-04-09T19:49:18.921Z root INFO PluginTheiaDirectoryHandler: accepting plugin with path
2023-04-09T19:49:18.921Z root INFO Resolved "vscode-language-pack-zh-hans" to a VS Code extension "vscode-language-pack-zh-hans@1.53.2" with engines:
2023-04-09T19:49:18.925Z daemon INFO INFO[0000] Using config file: /Users/cy/.arduinoIDE/arduino-cli.yaml
INFO[0000] arduino-cli version 0.31.0
INFO[0000] Executing `arduino-cli daemon`
{"IP":"127.0.0.1","Port":"53578"}

2023-04-09T19:49:18.925Z daemon INFO Daemon is running.
2023-04-09T19:49:18.927Z config INFO Loaded CLI configuration: {"board_manager":{"additional_urls":[]},"build_cache":{"compilations_before_purge":10,"ttl":"720h0m0s"},"daemon":{"port":"50051"},"directories":{"data":"/Users/cy/Library/Arduino15","downloads":"/Users/cy/Library/Arduino15/staging","user":"/Users/cy/Documents/Arduino"},"library":{"enable_unsafe_install":false},"logging":{"file":"","format":"text","level":"info"},"metrics":{"addr":":9090","enabled":true},"output":{"no_color":false},"sketch":{"always_export_binaries":false},"updater":{"enable_notification":true}}
2023-04-09T19:49:18.927Z config INFO 'directories.data' and 'directories.user' are set in the CLI configuration model.
2023-04-09T19:49:18.927Z config INFO Loaded the CLI configuration.
2023-04-09T19:49:18.931Z config INFO Mapped the CLI configuration: {"dataDirUri":"file:///Users/cy/Library/Arduino15","sketchDirUri":"file:///Users/cy/Documents/Arduino","additionalUrls":[],"network":"none","locale":"en"}
2023-04-09T19:49:18.931Z config INFO Validating the CLI configuration...
2023-04-09T19:49:18.936Z daemon INFO INFO[0000] Loading hardware from: /Users/cy/Library/Arduino15/packages

2023-04-09T19:49:18.936Z daemon INFO INFO[0000] Loading package arduino from: /Users/cy/Library/Arduino15/packages/arduino/hardware

2023-04-09T19:49:18.937Z daemon INFO INFO[0000] Checking signature                            error="opening signature file: open /Users/cy/Library/Arduino15/packages/arduino/hardware/avr/1.8.6/installed.json.sig: no such file or directory" index=/Users/cy/Library/Arduino15/packages/arduino/hardware/avr/1.8.6/installed.json signatureFile=/Users/cy/Library/Arduino15/packages/arduino/hardware/avr/1.8.6/installed.json.sig

2023-04-09T19:49:18.938Z daemon INFO INFO[0000] Adding monitor tool                           protocol=serial tool="builtin:serial-monitor"
INFO[0000] Loaded platform                               platform="arduino:avr@1.8.6"

2023-04-09T19:49:18.939Z root INFO Resolve plugins list: 40.7 ms [Finished 0.415 s after backend start]
2023-04-09T19:49:18.941Z root INFO Deployed backend plugin "marus25.cortex-debug@1.5.1" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/cortex-debug/extension/dist/extension.js": 1.8 ms [Finished 0.417 s after backend start]
2023-04-09T19:49:18.941Z daemon INFO INFO[0000] Checking signature                            error="opening signature file: open /Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_nano/4.0.2/installed.json.sig: no such file or directory" index=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_nano/4.0.2/installed.json signatureFile=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_nano/4.0.2/installed.json.sig
INFO[0000] Adding monitor tool                           protocol=serial tool="builtin:serial-monitor"
INFO[0000] Loaded platform                               platform="arduino:mbed_nano@4.0.2"

2023-04-09T19:49:18.943Z root INFO Deployed backend plugin "arduino.vscode-arduino-tools@0.0.2-beta.7" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-arduino-tools/extension/dist/bundle": 1.3 ms [Finished 0.418 s after backend start]
2023-04-09T19:49:18.943Z daemon INFO INFO[0000] Checking signature                            error="opening signature file: open /Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_rp2040/4.0.2/installed.json.sig: no such file or directory" index=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_rp2040/4.0.2/installed.json signatureFile=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_rp2040/4.0.2/installed.json.sig
INFO[0000] Adding monitor tool                           protocol=serial tool="builtin:serial-monitor"
INFO[0000] Loaded platform                               platform="arduino:mbed_rp2040@4.0.2"
INFO[0000] Checking existence of 'tools' path: /Users/cy/Library/Arduino15/packages/arduino/tools
INFO[0000] Loading tools from dir: /Users/cy/Library/Arduino15/packages/arduino/tools
INFO[0000] Loaded tool                                   tool="arduino:arduinoOTA@1.3.0"

2023-04-09T19:49:18.943Z daemon INFO INFO[0000] Loaded tool                                   tool="arduino:arm-none-eabi-gcc@7-2017q4"
INFO[0000] Loaded tool                                   tool="arduino:avr-gcc@7.3.0-atmel3.6.1-arduino7"
INFO[0000] Loaded tool                                   tool="arduino:avrdude@6.3.0-arduino17"

2023-04-09T19:49:18.943Z daemon INFO INFO[0000] Loaded tool                                   tool="arduino:bossac@1.9.1-arduino2"
INFO[0000] Loaded tool                                   tool="arduino:dfu-util@0.10.0-arduino1"

2023-04-09T19:49:18.947Z root INFO Deployed backend plugin "vscode.cpp@1.52.1" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-builtin-cpp/extension": 3.9 ms [Finished 0.422 s after backend start]
2023-04-09T19:49:18.947Z daemon INFO INFO[0000] Loaded tool                                   tool="arduino:openocd@0.11.0-arduino2"
INFO[0000] Loaded tool                                   tool="arduino:rp2040tools@1.0.6"
INFO[0000] Loading package builtin from: /Users/cy/Library/Arduino15/packages/builtin
INFO[0000] Checking existence of 'tools' path: /Users/cy/Library/Arduino15/packages/builtin/tools
INFO[0000] Loading tools from dir: /Users/cy/Library/Arduino15/packages/builtin/tools
INFO[0000] Loaded tool                                   tool="builtin:ctags@5.8-arduino11"
INFO[0000] Loaded tool                                   tool="builtin:mdns-discovery@1.0.9"
INFO[0000] Loaded tool                                   tool="builtin:serial-discovery@1.4.0"
INFO[0000] Loaded tool                                   tool="builtin:serial-monitor@0.13.0"

2023-04-09T19:49:18.947Z root INFO Deployed backend plugin "vscode.json@1.46.1" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-builtin-json/extension": 0.6 ms [Finished 0.423 s after backend start]
2023-04-09T19:49:18.947Z daemon INFO INFO[0000] Checking signature                            index=/Users/cy/Library/Arduino15/package_index.json signatureFile=/Users/cy/Library/Arduino15/package_index.json.sig trusted=true
INFO[0000] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_rp2040/4.0.2/libraries location=platform

2023-04-09T19:49:18.947Z daemon INFO INFO[0000] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/packages/arduino/hardware/avr/1.8.6/libraries location=platform

2023-04-09T19:49:18.948Z daemon INFO INFO[0000] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_nano/4.0.2/libraries location=platform
INFO[0000] Loading libraries index file                  index=/Users/cy/Library/Arduino15/library_index.json

2023-04-09T19:49:18.948Z root INFO Deployed backend plugin "vscode.json-language-features@1.46.1" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-builtin-json-language-features/extension/client/dist/jsonMain": 0.4 ms [Finished 0.423 s after backend start]
2023-04-09T19:49:18.976Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-bg@1.48.3" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-bg/extension": 28.6 ms [Finished 0.452 s after backend start]
2023-04-09T19:49:18.976Z config INFO The CLI config is valid.
2023-04-09T19:49:18.976Z config INFO <<< Initialized the CLI configuration.
2023-04-09T19:49:18.998Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-cs@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-cs/extension": 21.3 ms [Finished 0.473 s after backend start]
Restoring workspace roots: /Users/cy/Documents/Arduino/mic
isTempSketch: false. Input was /Users/cy/Documents/Arduino/mic/mic.ino
2023-04-09T19:49:19.036Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-de@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-de/extension": 38.0 ms [Finished 0.511 s after backend start]
2023-04-09T19:49:19.036Z daemon INFO INFO[0000] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/libraries location=ide
INFO[0000] Adding libraries dir                          dir=/Users/cy/Documents/Arduino/libraries location=user

2023-04-09T19:49:19.072Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-es@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-es/extension": 35.8 ms [Finished 0.547 s after backend start]
2023-04-09T19:49:19.076Z discovery-log INFO start request start watch
2023-04-09T19:49:19.076Z discovery-log INFO start requested start watch
2023-04-09T19:49:19.076Z discovery-log INFO start resolved watching
2023-04-09T19:49:19.077Z daemon INFO INFO[0000] starting discovery builtin:mdns-discovery process
INFO[0000] starting discovery builtin:serial-discovery process

2023-04-09T19:49:19.094Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-fr@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-fr/extension": 23.0 ms [Finished 0.570 s after backend start]
2023-04-09T19:49:19.095Z daemon INFO INFO[0000] started discovery builtin:mdns-discovery process
INFO[0000] sending command HELLO 1 "arduino-cli 0.31.0" to discovery builtin:mdns-discovery
INFO[0000] started discovery builtin:serial-discovery process
INFO[0000] sending command HELLO 1 "arduino-cli 0.31.0" to discovery builtin:serial-discovery
INFO[0000] from discovery builtin:serial-discovery received message type: hello, message: OK, protocol version: 1
INFO[0000] sending command START_SYNC to discovery builtin:serial-discovery
INFO[0000] from discovery builtin:serial-discovery received message type: start_sync, message: OK
INFO[0000] from discovery builtin:mdns-discovery received message type: hello, message: OK, protocol version: 1
INFO[0000] sending command START_SYNC to discovery builtin:mdns-discovery
INFO[0000] from discovery builtin:serial-discovery received message type: add, port: /dev/cu.Bluetooth-Incoming-Port
INFO[0000] from discovery builtin:mdns-discovery received message type: start_sync, message: OK

2023-04-09T19:49:19.096Z discovery-log INFO {"eventType":"add","port":{"matchingBoardsList":[],"port":{"address":"/dev/cu.Bluetooth-Incoming-Port","label":"/dev/cu.Bluetooth-Incoming-Port","protocol":"serial","protocolLabel":"Serial Port","propertiesMap":[],"hardwareId":""}},"error":""}
2023-04-09T19:49:19.108Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-hu@1.48.3" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-hu/extension": 13.9 ms [Finished 0.584 s after backend start]
2023-04-09T19:49:19.125Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-it@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-it/extension": 17.0 ms [Finished 0.601 s after backend start]
2023-04-09T19:49:19.143Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-ja@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-ja/extension": 17.8 ms [Finished 0.619 s after backend start]
2023-04-09T19:49:19.158Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-ko@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-ko/extension": 14.7 ms [Finished 0.634 s after backend start]
2023-04-09T19:49:19.164Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-nl@1.48.3" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-nl/extension": 6.2 ms [Finished 0.640 s after backend start]
2023-04-09T19:49:19.178Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-pl@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-pl/extension": 14.1 ms [Finished 0.654 s after backend start]
2023-04-09T19:49:19.194Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-pt-br@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-pt-BR/extension": 15.8 ms [Finished 0.670 s after backend start]
2023-04-09T19:49:19.213Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-ru@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-ru/extension": 18.6 ms [Finished 0.688 s after backend start]
2023-04-09T19:49:19.227Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-tr@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-tr/extension": 14.1 ms [Finished 0.702 s after backend start]
2023-04-09T19:49:19.239Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-uk@1.48.3" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-uk/extension": 11.6 ms [Finished 0.714 s after backend start]
2023-04-09T19:49:19.256Z root INFO Deployed backend plugin "ms-ceintl.vscode-language-pack-zh-hans@1.53.2" from "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-language-pack-zh-hans/extension": 15.1 ms [Finished 0.729 s after backend start]
2023-04-09T19:49:19.256Z root INFO Deploy plugins list: 40.7 ms [Finished 0.732 s after backend start]
Opening channel for service path '/services/electron-window-ext'.
Opening channel for service path '/services/electron-window'.
Opening channel for service path '/services/ide-updater'.
2023-04-09T19:49:19.734Z root WARN Frontend a.configure took longer than the expected maximum 100 milliseconds: 158.5 ms [Finished 0.695 s after frontend start]
2023-04-09T19:49:19.822Z sketches-service WARN Warning! The request sketch path was different than the response sketch path from the CLI. This could be a potential bug. Request: </Users/cy/Documents/Arduino/mic/mic.ino>, response: </Users/cy/Documents/Arduino/mic>.
2023-04-09T19:49:19.844Z root INFO Start frontend contributions: 299.0 ms [Finished 0.806 s after frontend start]
2023-04-09T19:49:19.844Z root INFO Changed application state from 'init' to 'started_contributions'.
2023-04-09T19:49:19.862Z root INFO Changed application state from 'started_contributions' to 'attached_shell'.
2023-04-09T19:49:19.862Z root INFO >>> Restoring the layout state...
2023-04-09T19:49:19.862Z root INFO ------- SERIALIZED LAYOUT DATA -------
2023-04-09T19:49:19.862Z root INFO {"version":5,"mainPanel":{"main":{"type":"tab-area","widgets":[{"constructionOptions":{"factoryId":"code-editor-opener","options":{"counter":0,"kind":"navigatable","uri":"file:///Users/cy/Documents/Arduino/mic/mic.ino"}},"innerWidgetState":"{\"cursorState\":[{\"inSelectionMode\":false,\"selectionStart\":{\"lineNumber\":1,\"column\":1},\"position\":{\"lineNumber\":1,\"column\":1}}],\"viewState\":{\"scrollLeft\":0,\"firstPosition\":{\"lineNumber\":1,\"column\":1},\"firstPositionDeltaTop\":0},\"contributionsState\":{\"editor.contrib.folding\":{\"lineCount\":37,\"provider\":\"indent\",\"foldedImports\":false},\"editor.contrib.wordHighlighter\":false}}"}],"currentIndex":0}},"mainPanelPinned":[false],"bottomPanel":{"config":{"main":{"type":"tab-area","widgets":[{"constructionOptions":{"factoryId":"outputView"},"innerWidgetState":"{\"locked\":false}"}],"currentIndex":0}},"pinned":[false],"size":241,"expanded":true},"leftPanel":{"type":"sidepanel","items":[{"widget":{"constructionOptions":{"factoryId":"arduino-sketchbook-widget"}},"rank":1,"expanded":false,"pinned":false},{"widget":{"constructionOptions":{"factoryId":"boards-list-widget"}},"rank":2,"expanded":true,"pinned":false},{"widget":{"constructionOptions":{"factoryId":"library-list-widget"}},"rank":3,"expanded":false,"pinned":false},{"widget":{"constructionOptions":{"factoryId":"debug"},"innerWidgetState":"{\"parts\":[{\"partId\":\"debug:threads:-1\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1.384,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"debug:threads\"},\"innerWidgetState\":\"{}\"}},{\"partId\":\"debug:frames:-1\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1.384,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"debug:frames\"},\"innerWidgetState\":\"{}\"}},{\"partId\":\"debug:variables:-1\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1.384,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"debug:variables\"},\"innerWidgetState\":\"{}\"}},{\"partId\":\"debug:watch:-1\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1.384,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"debug:watch\"},\"innerWidgetState\":\"{}\"}},{\"partId\":\"debug:breakpoints:-1\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1.384,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"debug:breakpoints\"},\"innerWidgetState\":\"{}\"}},{\"partId\":\"plugin-view:cortex-debug.peripherals\",\"collapsed\":true,\"hidden\":true,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"plugin-view\",\"options\":{\"id\":\"plugin-view:cortex-debug.peripherals\",\"viewId\":\"cortex-debug.peripherals\"}},\"innerWidgetState\":\"{\\\"label\\\":\\\"Cortex Peripherals\\\",\\\"widgets\\\":[],\\\"suppressUpdateViewVisibility\\\":false,\\\"currentViewContainerId\\\":\\\"debug\\\"}\"}},{\"partId\":\"plugin-view:cortex-debug.registers\",\"collapsed\":true,\"hidden\":true,\"originalContainerId\":\"debug:view-container:-1\",\"originalContainerTitle\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"plugin-view\",\"options\":{\"id\":\"plugin-view:cortex-debug.registers\",\"viewId\":\"cortex-debug.registers\"}},\"innerWidgetState\":\"{\\\"label\\\":\\\"Cortex Registers\\\",\\\"widgets\\\":[],\\\"suppressUpdateViewVisibility\\\":false,\\\"currentViewContainerId\\\":\\\"debug\\\"}\"}}],\"title\":{\"label\":\"debug\",\"iconClass\":\"codicon codicon-debug-alt\",\"closeable\":true}}"},"rank":4,"expanded":false,"pinned":false},{"widget":{"constructionOptions":{"factoryId":"search-view-container"},"innerWidgetState":"{\"parts\":[{\"partId\":\"search-in-workspace\",\"collapsed\":false,\"hidden\":false,\"relativeSize\":1,\"originalContainerId\":\"search-view-container\",\"originalContainerTitle\":{\"label\":\"Search\",\"iconClass\":\"codicon codicon-search\",\"closeable\":true},\"widget\":{\"constructionOptions\":{\"factoryId\":\"search-in-workspace\"},\"innerWidgetState\":\"{\\\"matchCaseState\\\":{\\\"className\\\":\\\"codicon codicon-case-sensitive\\\",\\\"enabled\\\":false,\\\"title\\\":\\\"Match Case\\\"},\\\"wholeWordState\\\":{\\\"className\\\":\\\"codicon codicon-whole-word\\\",\\\"enabled\\\":false,\\\"title\\\":\\\"Match Whole Word\\\"},\\\"regExpState\\\":{\\\"className\\\":\\\"codicon codicon-regex\\\",\\\"enabled\\\":false,\\\"title\\\":\\\"Use Regular Expression\\\"},\\\"includeIgnoredState\\\":{\\\"className\\\":\\\"codicon codicon-eye\\\",\\\"enabled\\\":false,\\\"title\\\":\\\"Include Ignored Files\\\"},\\\"showSearchDetails\\\":false,\\\"searchInWorkspaceOptions\\\":{\\\"matchCase\\\":false,\\\"matchWholeWord\\\":false,\\\"useRegExp\\\":false,\\\"includeIgnored\\\":false,\\\"include\\\":[],\\\"exclude\\\":[],\\\"maxResults\\\":2000},\\\"searchTerm\\\":\\\"\\\",\\\"replaceTerm\\\":\\\"\\\",\\\"showReplaceField\\\":false,\\\"searchHistoryState\\\":{\\\"history\\\":[\\\"Espressif32\\\"],\\\"index\\\":0},\\\"replaceHistoryState\\\":{\\\"history\\\":[],\\\"index\\\":0},\\\"includeHistoryState\\\":{\\\"history\\\":[],\\\"index\\\":0},\\\"excludeHistoryState\\\":{\\\"history\\\":[],\\\"index\\\":0}}\"}}],\"title\":{\"label\":\"Search\",\"iconClass\":\"fa fa-arduino-search\",\"closeable\":true}}"},"rank":5,"expanded":false,"pinned":false}],"size":248},"rightPanel":{"type":"sidepanel","items":[]}}
2023-04-09T19:49:19.863Z root INFO ------- END SERIALIZED LAYOUT DATA -------
2023-04-09T19:49:19.874Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070] Waiting for backend deployment: 38.7 ms [Finished 0.843 s after frontend start]
2023-04-09T19:49:20.050Z root INFO Refreshing token google-oauth2|101192729192564391508
2023-04-09T19:49:20.051Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286) starting instance
2023-04-09T19:49:20.213Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070] Sync of 21 plugins: 340.1 ms [Finished 1.183 s after frontend start]
2023-04-09 21:49:20.257 Arduino IDE Helper (Renderer)[2273:28804] CoreText note: Client requested name ".NewYork-Regular", it will get TimesNewRomanPSMT rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2023-04-09 21:49:20.257 Arduino IDE Helper (Renderer)[2273:28804] CoreText note: Set a breakpoint on CTFontLogSystemFontNameRequest to debug.
2023-04-09T19:49:20.358Z root INFO <<< The layout has been successfully restored.
2023-04-09T19:49:20.400Z root INFO Initialize the workbench layout: 502.2 ms [Finished 1.329 s after frontend start]
2023-04-09T19:49:20.401Z root INFO Changed application state from 'attached_shell' to 'initialized_layout'.
2023-04-09T19:49:20.401Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][marus25.cortex-debug]: Loaded contributions.
2023-04-09T19:49:20.401Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][arduino.vscode-arduino-tools]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.cpp]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.json]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.json-language-features]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-bg]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-cs]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-de]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-es]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-fr]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-hu]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-it]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ja]: Loaded contributions.
2023-04-09T19:49:20.402Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ko]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-nl]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-pl]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-pt-br]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ru]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-tr]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-uk]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-zh-hans]: Loaded contributions.
2023-04-09T19:49:20.403Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070] Load contributions of 21 plugins: 40.7 ms [Finished 1.370 s after frontend start]
2023-04-09T19:49:20.424Z root WARN 'cpp' language is remapped from 'source.cpp.embedded.macro' to 'source.cpp' scope
2023-04-09T19:49:20.424Z root WARN a registered grammar configuration for 'cpp' language is overridden
2023-04-09T19:49:20.481Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): PluginManagerExtImpl/init()
2023-04-09T19:49:20.482Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(cortex-debug@1.5.1 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.484Z root INFO [hosted-plugin: 2286] Debugger contribution has been registered: cortex-debug
2023-04-09T19:49:20.484Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-arduino-tools@0.0.2-beta.7 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.484Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(cpp@1.52.1 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.485Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(json@1.46.1 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.485Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(json-language-features@1.46.1 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.485Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-bg@1.48.3 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.486Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-cs@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.486Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-de@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.486Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-es@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.487Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-fr@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.488Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-hu@1.48.3 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.488Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-it@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.488Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-ja@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.488Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-ko@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.489Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-nl@1.48.3 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.489Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-pl@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.489Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-pt-BR@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.489Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-ru@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.490Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-tr@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.491Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): initializing(vscode-language-pack-uk@1.48.3 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
PLUGIN_HOST(2286): initializing(vscode-language-pack-zh-hans@1.53.2 with /Applications/Arduino IDE.app/Contents/Resources/app/node_modules/@theia/plugin-ext-vscode/lib/node/plugin-vscode-init)
2023-04-09T19:49:20.638Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): PluginManagerExtImpl/loadPlugin(/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-builtin-json-language-features/extension/client/dist/jsonMain)
2023-04-09T19:49:20.658Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): PluginManagerExtImpl/loadPlugin(/Applications/Arduino IDE.app/Contents/Resources/app/plugins/vscode-arduino-tools/extension/dist/bundle)
2023-04-09T19:49:20.670Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][marus25.cortex-debug]: Started plugin.
2023-04-09T19:49:20.670Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][arduino.vscode-arduino-tools]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.cpp]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.json]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][vscode.json-language-features]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-bg]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-cs]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-de]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-es]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-fr]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-hu]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-it]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ja]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ko]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-nl]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-pl]: Started plugin.
2023-04-09T19:49:20.671Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-pt-br]: Started plugin.
2023-04-09T19:49:20.672Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-ru]: Started plugin.
2023-04-09T19:49:20.672Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-tr]: Started plugin.
2023-04-09T19:49:20.672Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-uk]: Started plugin.
2023-04-09T19:49:20.672Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070][ms-ceintl.vscode-language-pack-zh-hans]: Started plugin.
2023-04-09T19:49:20.673Z root INFO [hosted-plugin: 2286] PLUGIN_HOST(2286): PluginManagerExtImpl/loadPlugin(/Applications/Arduino IDE.app/Contents/Resources/app/plugins/cortex-debug/extension/dist/extension.js)
2023-04-09T19:49:20.687Z root INFO [hosted-plugin: 2286] Debug configuration provider has been registered: cortex-debug, trigger: 1
2023-04-09T19:49:20.687Z root INFO [hosted-plugin: 2286] GDB server console created
2023-04-09T19:49:20.691Z root INFO [e1b4aa8b-1b5c-4249-a20b-b1ff4c5d0070] Start of 21 plugins: 288.4 ms [Finished 1.661 s after frontend start]
2023-04-09T19:49:21.415Z root INFO Frontend application start: 1691.2 ms [Finished 2.891 s after backend start]
2023-04-09T19:49:21.417Z root INFO Replace loading indicator with ready workbench UI (animation): 824.3 ms [Finished 2.195 s after frontend start]
2023-04-09T19:49:21.419Z root INFO Changed application state from 'initialized_layout' to 'ready'.
2023-04-09T19:49:21.419Z root WARN Linked preference "workbench.colorCustomizations" not found. Source: "editor.bracketPairColorization.enabled"
Checking for update
Update for version 2.0.4 is not available (latest version: 2.0.4, downgrade is allowed).
2023-04-09T19:49:22.081Z root ERROR ---- !!!SLOW!!! DURATION: search took 433.535 ms. Args: [{"type":"Updatable"}] ----
2023-04-09T19:49:22.082Z root INFO Starting language server: arduino:mbed_nano:nanorp2040connect
2023-04-09T19:49:22.142Z daemon INFO INFO[0003] Required tool                                 tool="arduino:arm-none-eabi-gcc@7-2017q4"
INFO[0003] Required tool                                 tool="arduino:bossac@1.9.1-arduino2"
INFO[0003] Required tool                                 tool="arduino:dfu-util@0.10.0-arduino1"
INFO[0003] Required tool                                 tool="arduino:openocd@0.11.0-arduino2"
INFO[0003] Required tool                                 tool="arduino:rp2040tools@1.0.6"

2023-04-09T19:49:22.142Z daemon INFO INFO[0003] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/libraries location=ide
INFO[0003] Adding libraries dir                          dir=/Users/cy/Library/Arduino15/packages/arduino/hardware/mbed_nano/4.0.2/libraries location=platform
INFO[0003] Adding libraries dir                          dir=/Users/cy/Documents/Arduino/libraries location=user

2023-04-09T19:49:22.171Z root ERROR Child node 'WorkspaceNodeId' does not belong to this 'WorkspaceNodeId' tree.
2023-04-09T19:49:22.467Z daemon INFO INFO[0003] Required tool                                 tool="arduino:arm-none-eabi-gcc@7-2017q4"
INFO[0003] Required tool                                 tool="arduino:bossac@1.9.1-arduino2"
INFO[0003] Required tool                                 tool="arduino:dfu-util@0.10.0-arduino1"

2023-04-09T19:49:22.467Z daemon INFO INFO[0003] Required tool                                 tool="arduino:openocd@0.11.0-arduino2"
INFO[0003] Required tool                                 tool="arduino:rp2040tools@1.0.6"

`

**Comment by grandinquisitor** [comment]
I found that I was able to work around this issue only by making sure my board was plugged in BEFORE starting the IDE. If I plugged it in after the IDE was started it never found it.
